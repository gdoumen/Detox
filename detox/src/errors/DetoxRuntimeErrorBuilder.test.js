const DetoxRuntimeErrorBuilder = require('./DetoxRuntimeErrorBuilder');

describe('DetoxRuntimeErrorBuilder', () => {
  /** @type () => DetoxRuntimeErrorBuilder */
  let builder;
  let appsConfig;

  beforeEach(() => {
    builder = () => new DetoxRuntimeErrorBuilder({ appsConfig });
  });

  test('abortedDetoxInit', () => {
    expect(builder().abortedDetoxInit()).toMatchSnapshot();
  });

  test('invalidTestSummary - beforeEach', () => {
    expect(builder().invalidTestSummary('beforeEach', 'deviceName')).toMatchSnapshot();
  });

  test('invalidTestSummary - afterEach', () => {
    expect(builder().invalidTestSummary('afterEach', { weird: [['string']] })).toMatchSnapshot();
  });

  test('invalidTestSummaryStatus - afterEach', () => {
    expect(builder().invalidTestSummary('afterEach', {
      title: 'foo test',
      fullName: 'suite foo test',
      status: 'uncertain',
    })).toMatchSnapshot();
  });

  test('cantFindApp - single app', () => {
    appsConfig = {
      default: {
        type: 'ios.app',
        binaryPath: 'path/to/app',
      }
    };

    expect(builder().cantFindApp('unicorn')).toMatchSnapshot();
  });

  test('cantFindApp - multiple apps', () => {
    appsConfig = {
      fish: {},
      chips: {},
      hummus: {},
      beer: {},
    };

    expect(builder().cantFindApp('steak')).toMatchSnapshot();
  });
});
