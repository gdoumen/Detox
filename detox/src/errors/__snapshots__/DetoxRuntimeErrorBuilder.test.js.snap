// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetoxRuntimeErrorBuilder abortedDetoxInit 1`] = `
[DetoxRuntimeError: Aborted detox.init() execution, and now running detox.cleanup()

HINT: Most likely, your test runner is tearing down the suite due to the timeout error]
`;

exports[`DetoxRuntimeErrorBuilder cantFindApp - multiple apps 1`] = `
[DetoxRuntimeError: Can't find an app config with name "steak".

HINT: Maybe you meant one of these app names?
* fish
* chips
* hummus
* beer]
`;

exports[`DetoxRuntimeErrorBuilder cantFindApp - single app 1`] = `
[DetoxRuntimeError: Can't find an app config with name "unicorn".

HINT: Actually, you don't have multiple apps defined in your config.
Hence, either you don't need to call device.selectApp("unicorn") at all,
or, your apps config is missing that app. See the actual contents:

{
  default: {
    type: 'ios.app',
    binaryPath: 'path/to/app'
  }
}]
`;

exports[`DetoxRuntimeErrorBuilder invalidTestSummary - afterEach 1`] = `
[DetoxRuntimeError: Invalid test summary was passed to detox.afterEach(testSummary)
Expected to get an object of type: { title: string; fullName: string; status: "running" | "passed" | "failed"; }

HINT: Maybe you are still using an old undocumented signature detox.beforeEach(string, string, string) in init.js ?
See the article for the guidance: https://github.com/wix/detox/blob/master/docs/APIRef.TestLifecycle.md
testSummary was: 

{
  weird: [Array]
}]
`;

exports[`DetoxRuntimeErrorBuilder invalidTestSummary - beforeEach 1`] = `
[DetoxRuntimeError: Invalid test summary was passed to detox.beforeEach(testSummary)
Expected to get an object of type: { title: string; fullName: string; status: "running" | "passed" | "failed"; }

HINT: Maybe you are still using an old undocumented signature detox.beforeEach(string, string, string) in init.js ?
See the article for the guidance: https://github.com/wix/detox/blob/master/docs/APIRef.TestLifecycle.md
testSummary was: 

deviceName]
`;

exports[`DetoxRuntimeErrorBuilder invalidTestSummaryStatus - afterEach 1`] = `
[DetoxRuntimeError: Invalid test summary was passed to detox.afterEach(testSummary)
Expected to get an object of type: { title: string; fullName: string; status: "running" | "passed" | "failed"; }

HINT: Maybe you are still using an old undocumented signature detox.beforeEach(string, string, string) in init.js ?
See the article for the guidance: https://github.com/wix/detox/blob/master/docs/APIRef.TestLifecycle.md
testSummary was: 

{
  title: 'foo test',
  fullName: 'suite foo test',
  status: 'uncertain'
}]
`;
